clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -500,17 +517,22 @@ public void onFailure(RuntimeException e) {
      *             or to any of the specified partitions
      * @throws CommitFailedException if an unrecoverable error occurs before the commit can be completed
      */
-    public void commitOffsetsSync(Map<TopicPartition, OffsetAndMetadata> offsets) {
+    public void commitOffsetsSync(Map<TopicPartition, OffsetAndMetadata> offsets, long timeoutMs) {
         invokeCompletedOffsetCommitCallbacks();
 
         if (offsets.isEmpty())
             return;
 
-        while (true) {
-            ensureCoordinatorReady();
+        long startMs = time.milliseconds();
+        long remainingMs = timeoutMs;
+        do {
+            ensureCoordinatorReady(remainingMs);
 
             RequestFuture<Void> future = sendOffsetCommitRequest(offsets);
-            client.poll(future);
+            if (timeoutMs == Long.MAX_VALUE)
+                client.poll(future);
+            else
+                client.poll(future, remainingMs);
```

Comment:
Do we need to recompute `remainingMs` after the call to `ensureCoordinatorReady` above?